test/SSHRemoting/SSHRemoting.Basic.Tests.ps1

```diff
@@ -6,72 +6,191 @@ Describe "SSHRemoting Basic Tests" -tags CI {
+        {
+            $connect.Add('ConnectingTimeout', $script:TestConnectingTimeout)
+        }
+
+        # Try creating a new SSH connection
+        $connectionError = $null
+        $session = $null
+        $count = 0
+        while (($null -eq $session) -and ($count++ -lt 2))
+        {
+            $session = New-PSSession @PSBoundParameters -ErrorVariable connectionError -ErrorAction SilentlyContinue
+            if ($null -eq $session)
+            {
+                Write-Verbose -Verbose "SSH New-PSSession remoting connect failed after $($timeout/1000) second wait."
+
+                if ($count -eq 1)
+                {
+                    # Try restarting sshd service
+                    CheckSSHDService
+                }
+            }
+        }
+
+        if ($null -eq $session)
+        {
+            $message = "New-PSSession unable to connect to SSH remoting endpoint after two attempts. Error: $($connectionError.Exception.Message)"
+            throw [System.Management.Automation.PSInvalidOperationException]::new($message)
+        }
+
+        Write-Verbose -Verbose "SSH New-PSSession remoting connect succeeded."
+        Write-Output $session
+    }
+
     function VerifySession {
         param (
             [System.Management.Automation.Runspaces.PSSession] $session
         )
 
+        Write-Verbose -Verbose "VerifySession called for session: $($session.Id)"
+
         $session.State | Should -BeExactly 'Opened'
         $session.ComputerName | Should -BeExactly 'localhost'
         $session.Transport | Should -BeExactly 'SSH'
+        Write-Verbose -Verbose "Invoking whoami"
         Invoke-Command -Session $session -ScriptBlock { whoami } | Should -BeExactly $(whoami)
+        Write-Verbose -Verbose "Invoking PSSenderInfo"
         $psRemoteVersion = Invoke-Command -Session $session -ScriptBlock { $PSSenderInfo.ApplicationArguments.PSVersionTable.PSVersion }
         $psRemoteVersion.Major | Should -BeExactly $PSVersionTable.PSVersion.Major
         $psRemoteVersion.Minor | Should -BeExactly $PSVersionTable.PSVersion.Minor
+        Write-Verbose -Verbose "VerifySession complete"
     }
 
     Context "New-PSSession Tests" {
 
         AfterEach {
+            Write-Verbose -Verbose "Starting New-PSSession AfterEach"
             if ($script:session -ne $null) { Remove-PSSession -Session $script:session }
             if ($script:sessions -ne $null) { Remove-PSSession -Session $script:sessions }
+            Write-Verbose -Verbose "AfterEach complete"
         }
 
         It "Verifies new connection with implicit current User" {
-            $script:session = New-PSSession -HostName localhost -ErrorVariable err
-            $err | Should -HaveCount 0
+            Write-Verbose -Verbose "It Starting: Verifies new connection with implicit current User"
+            $script:session = TryNewPSSession -HostName localhost
             VerifySession $script:session
+            Write-Verbose -Verbose "It Complete"
```

Comment:
It seems Pester "Should" is absent.

Reply:
The Pester 'Should' is called in the 'VerifySession' function.  Could that be what is causing the random hang?